src/order/rel_classes.lean

```diff
@@ -209,29 +209,38 @@ instance is_extensional_of_is_strict_total_order'
 
 /-! ### Well-order -/
 
-/-- A well order is a well-founded linear order. -/
-@[algebra] class is_well_order (α : Type u) (r : α → α → Prop)
-  extends is_strict_total_order' α r : Prop :=
+/-- A well-founded relation. Not to be confused with `is_well_order`. -/
+@[algebra] class is_well_founded (α : Type u) (r : α → α → Prop) : Prop :=
 (wf : well_founded r)
 
+instance has_well_founded.is_well_founded [h : has_well_founded α] :
+  is_well_founded α has_well_founded.r := { ..h }
+
+theorem is_well_founded.induction (r : α → α → Prop) [h : is_well_founded α r] {C : α → Prop} :
+  ∀ a, (∀ x, (∀ y, r y x → C y) → C x) → C a :=
+h.wf.induction
+
+theorem well_founded.asymmetric {α : Sort*} {r : α → α → Prop} (h : well_founded r) :
+  ∀ ⦃a b⦄, r a b → ¬ r b a
+| a := λ b hab hba, well_founded.asymmetric hba hab
+using_well_founded { rel_tac := λ _ _, `[exact ⟨_, h⟩],
+                     dec_tac := tactic.assumption }
+
 @[priority 100] -- see Note [lower instance priority]
-instance is_well_order.is_strict_total_order {α} (r : α → α → Prop) [is_well_order α r] :
```

Comment:
I don't know why these redundant typeclasses were here. I presume it's to short-circuit the search, but is this standard practice? If so, I'll add these again.